react-app/src/components/table/DataTable.tsx

```diff
@@ -298,3 +299,57 @@ export const FilterSearchDataGrid = (props: FilterSearchDataGridProps) => {
     </>
   );
 };
+
+type PinnedColumnDataGridProps = {
+  pinnedFields: string[];
+  pinnedSxProps?: SxProps;
+  scrollableSxProps?: SxProps;
+} & DataGridProps;
+
+/**
+ * This is a somewhat hacky workaround for pinned columns in the community version of mui-x.
+ * If we ever get a Pro sub, we can just get rid of this entirely.
+ * All the sorting and filtering options are disabled since this is just two data grids smushed together
+ * and their states are not synced at all.
+ */
```

Comment:
I see what you mean, but I'm not sure there's another way around it with MUI for now. Good workaround.